internal/bminventory/inventory.go

```diff
@@ -795,6 +795,10 @@ func (b *bareMetalInventory) getNewClusterCPUArchitecture(newClusterParams *mode
 		return "", errors.Errorf("Non x86_64 CPU architectures for version %s are supported only with User Managed Networking", swag.StringValue(newClusterParams.OpenshiftVersion))
 	}
 
+	if getPlatformType(newClusterParams.Platform) == string(models.PlatformTypeNutanix) && newClusterParams.CPUArchitecture == common.ARM64CPUArchitecture {
```

Comment:
Is using `isArm64AllowedForPlatform` here possible?

Reply:
Rework using @slaviered's suggestion as `validations.ValidateArchitectureWithPlatform` in DRY spirit